pages/api/chat.api.ts

```diff
@@ -27,32 +27,28 @@ function errorResponse(body: any, status: number) {
 
 const handler = async (req: Request): Promise<Response> => {
   try {
-    const encoding = await getTiktokenEncoding()
-    const {messages, apiKey, modelId, prompt, temperature} = (await req.json()) as ChatBody
+    const encoder = await TiktokenEncoder.create()
```

Comment:
This call takes really long (several seconds), and it may be one of the reasons why it takes long before you get an answer after pressing Enter. Maybe we can avoid calculating the number of tokens every time you press Enter? We already know how many tokens there are (as we calculate it every 3 seconds for the alert next to the prompt.)